### PR TITLE
fix stale remote cluster uuid state not purged from remote

### PR DESCRIPTION
**Pull request by linuxpi**
### Description
[Describe what this change achieves]

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java [hunk 1]
```diff
@@ -661,30 +668,42 @@ private boolean isInvalidClusterUUID(ClusterMetadataManifest manifest) {
...
+     * Fetch ClusterMetadataManifest files from remote state store in order
      *
      * @param clusterUUID uuid of cluster state to refer to in remote
      * @param clusterName name of the cluster
-     * @return latest ClusterMetadataManifest filename
+     * @param limit max no of files to fetch
+     * @return all manifest file names
      */
-    private Optional<String> getLatestManifestFileName(String clusterName, String clusterUUID) throws IllegalStateException {
+    private List<BlobMetadata> getManifestFileNames(String clusterName, String clusterUUID, int limit) throws IllegalStateException {
```
Comment (soosinha) [inline server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java:678]: limit is not getting used

Comment (linuxpi) [reply server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java:678]: Thanks for catching this. Fixed it.

File: test/framework/src/main/java/org/opensearch/bootstrap/BootstrapForTesting.java [hunk 2]
```diff
@@ -171,8 +171,8 @@ public boolean implies(ProtectionDomain domain, Permission permission) {
                 });
                 // Create access control context for mocking
                 PriviledgedMockMaker.createAccessControlContext();
-                System.setSecurityManager(SecureSM.createTestSecureSM(getTrustedHosts()));
-                Security.selfTest();
+//                System.setSecurityManager(SecureSM.createTestSecureSM(getTrustedHosts()));
+//                Security.selfTest();
```
Comment (sachinpkale) [inline test/framework/src/main/java/org/opensearch/bootstrap/BootstrapForTesting.java:175]: If we don't need these lines, let's remove them.

Comment (linuxpi) [reply test/framework/src/main/java/org/opensearch/bootstrap/BootstrapForTesting.java:175]: yes, it was pushed by mistake, corrected this.

**Review by sachinpkale — APPROVED** [review]

**Review by sachinpkale — APPROVED** [review]

**Pull request merged**